### PR TITLE
Nano: add dGPU first push for pytorch InferenceOptimizer

### PR DESCRIPTION
**Pull request by TheaperDeng**
## Description

**This PR needs manual check before merging due to the lack of CI/CD hardware**

### 1. Why the change?
This PR is a first push for dGPU support for pytorch InferenceOptimizer

### 2. User API changes
dGPU support is still a "**dependency-heavy**" one. This PR provides some solution while need some revise in future.

| Requirement | Sub-reqirement   | Current status                            | Comment                                                                  |
| ----------- | ---------------- | ----------------------------------------- | --------------------------------------------------------------------------- |
| Driver      |                  | ❌ Manual                                  | https://dgpu-docs.intel.com/installation-guides/ubuntu/ubuntu-jammy-dc.html |
| OneAPI      | Installation     | ✳️ Nano **check** by `bigdl-nano-init -g/--gpu` |                                                                             |
|             | Version          | ❌ Manual check                            |                                                                             |
|             | Variable setting | ✅ Nano **handle** by `bigdl-nano-init -g/--gpu` |                                                                             |
| IPEX        | torch-private    | ✅ Nano **handle** by `pip install`        |                                                                             |
|             | Ipex+xpu         | ✅ Nano **handle** by `pip install`        |                                                                             |

- Currently we only support 1.13 version, by `pip install --pre bigdl-nano[pytorch_113_xpu]`
- No fp16, ipex optimization for now, will add later

### Installation
```bash
pip install --pre bigdl-nano[pytorch_113_xpu] -f https://developer.intel.com/ipex-whl-stable-xpu
source bigdl-nano-init -g/--gpu
```

### Python API
```python
from bigdl.nano.pytorch import InferenceOptimizer

model = InferenceOptimizer.trace(model, device="GPU")
model(x)
```

### 3. Summary of the change 
1. add some changes to `bigdl-nano-init`, some OneAPI related var settings are added (add to nano_var.sh as well)
2. add a new class for dgpu transformation `PytorchIPEXXPUModel`
3. add a new option `pytorch_113_xpu` to setup

### 4. How to test?
- [x] Manual test
- [x] FP32 inference
- [x] FP32 save/load
- [x] FP32 + IPEX inference
- [x] FP32 + save/load
- [x] BF32 inference
- [x] BF32 save/load

### 5. New dependencies

- [x] New Python dependencies
       - torch==1.13.0a0
       - torchvision==0.14.1a0
       - intel_extension_for_pytorch==1.13.10+xpu

### CODE REVIEW

File: python/nano/src/bigdl/nano/deps/ipex/ipex_inference_xpu_model.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import intel_extension_for_pytorch as ipex
+from bigdl.nano.utils.pytorch import apply_data_to_xpu
+from bigdl.nano.pytorch.context_manager import generate_context_manager
+
+
+class PytorchIPEXPUModel(AcceleratedLightningModule):
+
+    def __init__(self, model: torch.nn.Module, thread_num=None):
+        super().__init__(model)
+        self.model = model.to('xpu')
```
Comment (MeouSker77) [inline python/nano/src/bigdl/nano/deps/ipex/ipex_inference_xpu_model.py:28]: Is this inplaced?

File: python/nano/src/bigdl/nano/deps/ipex/ipex_inference_xpu_model.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+#
+
+from bigdl.nano.pytorch.model import AcceleratedLightningModule
+import torch
+import intel_extension_for_pytorch as ipex
+from bigdl.nano.utils.pytorch import apply_data_to_xpu
+from bigdl.nano.pytorch.context_manager import generate_context_manager
+
+
+class PytorchIPEXPUModel(AcceleratedLightningModule):
```
Comment (MeouSker77) [inline python/nano/src/bigdl/nano/deps/ipex/ipex_inference_xpu_model.py:24]: For this wrapped model, we can use approach like this <https://github.com/intel-analytics/BigDL/blob/26125da1bbcdeead4bf78ed3295993cc9dcdd7f8/python/nano/src/bigdl/nano/pytorch/amp/bfloat16.py#L110> to support custom attributes access.

**Review by MeouSker77 — APPROVED** [review]
others LGTM

File: python/nano/scripts/bigdl-nano-init [hunk 3]
```diff
@@ -36,6 +36,14 @@ function enable-perf-var {
     PERF_VAR=1
 }
 
+function enable-dgpu-var {
+    echo "Option: Enable OneAPI vars"
+    ONEAPI_VAR=1
+    echo "Option: Disable jemalloc and tcmalloc and unset related variables"
+    unset ENABLE_JEMALLOC_VAR
+    unset ENABLE_TCMALLOC_VAR
```
[the remark this answers is not in the document]
Comment (TheaperDeng) [reply python/nano/scripts/bigdl-nano-init:44]: We could allow this behavior (maybe tcmalloc will be supported by oneapi in future :) )

File: python/nano/src/bigdl/nano/deps/ipex/ipex_inference_xpu_model.py (hunk 1 above)
Comment (TheaperDeng) [reply python/nano/src/bigdl/nano/deps/ipex/ipex_inference_xpu_model.py:42]: inplace issue will be considerred in next PR.

**Pull request merged**